spec/rubocop/cop/rails/http_positional_arguments_spec.rb

```diff
@@ -0,0 +1,227 @@
+    let(:source) do
+      'get :new, params: { user_id: @user.id }'
+    end
+
+    it 'does not register an offense' do
+      inspect_source(cop, source)
+      expect(cop.messages).to be_empty
+    end
+  end
+
+  describe :get do
+    let(:source) do
+      'get :new, user_id: @user.id'
+    end
+
+    let(:corrected_result) do
+      'get :new, params: { user_id: @user.id }'
+    end
+
+    it 'records an offense' do
+      inspect_source(cop, source)
+      expect(cop.offenses.size).to eq(1)
+    end
+
+    it 'autocorrects offense' do
+      new_source = autocorrect_source(cop, source)
+      expect(new_source).to eq(corrected_result)
+    end
+
+    describe 'no params' do
+      let(:source) do
+        'get :new'
+      end
+
+      it 'does not record an offense' do
+        inspect_source(cop, source)
+        expect(cop.offenses.size).to eq(0)
+      end
+    end
+  end
+
+  describe :patch do
+    let(:source) do
+      <<-EOS
+patch :update,
+          id: @user.id,
+          ac: {
+            article_id: @article1.id,
+            profile_id: @profile1.id,
+            content: 'Some Text'
+          }
+EOS
+    end
+
+    let(:corrected_result) do
+      <<-EOS
+patch :update, params: { id: @user.id, ac: {
+            article_id: @article1.id,
+            profile_id: @profile1.id,
+            content: 'Some Text'
+          } }
+EOS
+    end
+
+    it 'records an offense' do
+      inspect_source(cop, source)
+      expect(cop.offenses.size).to eq(1)
+    end
+
+    it 'autocorrects offense' do
+      new_source = autocorrect_source(cop, source)
+      expect(new_source).to eq(corrected_result)
+    end
+  end
+
+  describe :post do
+    let(:source) do
+      <<-EOS
+post :create,
+          id: @user.id,
+          ac: {
+            article_id: @article1.id,
+            profile_id: @profile1.id,
+            content: 'Some Text'
+          }
+      EOS
+    end
+
+    let(:corrected_result) do
+      <<-EOS
+post :create, params: { id: @user.id, ac: {
+            article_id: @article1.id,
+            profile_id: @profile1.id,
+            content: 'Some Text'
+          } }
+      EOS
+    end
+
+    it 'records an offense' do
+      inspect_source(cop, source)
+      expect(cop.offenses.size).to eq(1)
+    end
+
+    it 'autocorrects offense' do
+      new_source = autocorrect_source(cop, source)
+      expect(new_source).to eq(corrected_result)
+    end
+  end
+
+  %w(post get patch put delete).each do |keyword|
+    it 'does not record an offense when keyword found on variable' do
+      source = "@user.#{keyword}.id = ''"
+      inspect_source(cop, source)
+      expect(cop.offenses.size).to eq(0)
+    end
+
+    it 'does not record an offense when keyword is found on object' do
+      source = "obj.#{keyword}.profile.id"
+      inspect_source(cop, source)
+      expect(cop.offenses.size).to eq(0)
+    end
+  end
+
+  it 'auto corrects controller method when it is a method' do
```

Comment:
auto-corrects

I'd drop the term controller method, as those tests don't really check whether something is living inside a controller.